src/Model/CategoryManagerInterface.php

```diff
@@ -25,8 +25,10 @@
  * @method CategoryInterface[]    getRootCategoriesSplitByContexts(bool $loadChildren = true)
  * @method CategoryInterface[]    getCategories(ContextInterface|string|null $context)
  * @method CategoryInterface|null getBySlug(string $slug, ContextInterface|string|null $context, bool $enabled = true)
+ *
+ * @phpstan-extends ManagerInterface<CategoryInterface>
  */
-interface CategoryManagerInterface extends ManagerInterface, PageableManagerInterface
+interface CategoryManagerInterface extends ManagerInterface, PageableInterface
```

Comment:
The PageableInterface has 
```
@template-covariant T of object
```
so a `phpstan-extends PageableInterface< CategoryInterface >`
could also be added (same for others interface).

Reply:
Done